### PR TITLE
Feat/vsdspub 529/shacl shape update

### PR DESCRIPTION
**Pull request by jobulcke**
Use the shacl shape configured by the admin api to validate the ingested members

### CODE REVIEW

File: ldes-server-port-ingestion-rest/pom.xml [hunk 1]
```diff
@@ -12,6 +12,8 @@
 
     <properties>
         <relative-formatter-path>..</relative-formatter-path>
+        <cucumber.version>7.11.2</cucumber.version>
+        <junit-platform-suite.version>1.9.2</junit-platform-suite.version>
```
Comment (Yalz) [inline ldes-server-port-ingestion-rest/pom.xml:16]: Should this not be put on top level pom ?

**Review by Yalz — APPROVED** [review]

File: ldes-server-port-admin-rest/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/admin/rest/controllers/AdminShapeRestController.java [hunk 2]
```diff
@@ -19,6 +22,9 @@ public class AdminShapeRestController {
 	@Qualifier("shapeShaclValidator")
 	private LdesConfigShaclValidator shapeValidator;
 
+	@Autowired
+	private ApplicationEventPublisher eventPublisher;
+
 	@Autowired
```
Comment (WLefever-Cegeka) [inline ldes-server-port-admin-rest/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/admin/rest/controllers/AdminShapeRestController.java:28]: minor suggestion: get rid of @Autowired

File: ldes-server-port-admin-rest/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/admin/rest/controllers/AdminShapeRestController.java [hunk 3]
```diff
@@ -39,7 +45,14 @@ public ResponseEntity<LdesConfigModel> getShape(@PathVariable String collectionN
...
+		eventPublisher.publishEvent(new ShaclChangedEvent(collectionName, shape.getModel()));
 		return ResponseEntity.ok(updatedShape);
 	}
 
+	@PostConstruct
+	private void initShapeConfig() {
+		service.retrieveAllShapes().stream()
+				.map(configModel -> new ShaclChangedEvent(configModel.getId(), configModel.getModel()))
+				.forEach(eventPublisher::publishEvent);
+	}
```
Comment (WLefever-Cegeka) [inline ldes-server-port-admin-rest/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/admin/rest/controllers/AdminShapeRestController.java:57]: Like the idea, have a suggestion for an alternative implementation. Maybe we could have a look later. Made an issue: https://github.com/Informatievlaanderen/VSDS-LDESServer4J/issues/527

File: ldes-server-domain/src/test/java/be/vlaanderen/informatievlaanderen/ldes/server/domain/validation/LdesShaclValidatorTest.java [hunk 4]
```diff
@@ -61,6 +63,50 @@ void when_ValidateWithNoProvidedShape_thenReturnValid() throws URISyntaxExceptio
...
+
+		ldesShaclValidator = new LdesShaclValidator(config);
+
+		Member invalidMember = readLdesMemberFromFile("validation/example-data-invalid.ttl");
+		Model invalidModel = invalidMember.getModel();
+		assertThrows(LdesShaclValidationException.class, () -> ldesShaclValidator.validateShape(invalidModel));
+	}
+
+	@Test
+	void when_ValidateProvidedInvalidAta_and_OnlyDbShapeProvided_thenReturnInvalid()
```
Comment (WLefever-Cegeka) [inline ldes-server-domain/src/test/java/be/vlaanderen/informatievlaanderen/ldes/server/domain/validation/LdesShaclValidatorTest.java:99]: InvalidAta -> InvalidData

File: ldes-server-port-admin-rest/src/test/java/be/vlaanderen/informatievlaanderen/ldes/server/admin/rest/controllers/AdminShapeRestControllerTest.java [hunk 5]
```diff
@@ -44,13 +49,17 @@
...
+@RecordApplicationEvents
 class AdminShapeRestControllerTest {
 	@MockBean
 	private LdesConfigModelService ldesConfigModelService;
 	@MockBean
 	@Qualifier("shapeShaclValidator")
 	private LdesConfigShaclValidator ldesConfigShaclValidator;
 
+	@Autowired
+	private ApplicationEvents applicationEvents;
```
Comment (WLefever-Cegeka) [inline ldes-server-port-admin-rest/src/test/java/be/vlaanderen/informatievlaanderen/ldes/server/admin/rest/controllers/AdminShapeRestControllerTest.java:61]: Ha, that's a secret trick. I made a suggestion using a captor: https://www.baeldung.com/mockito-argumentcaptor

**Review by WLefever-Cegeka — APPROVED** [review]

**Pull request merged**